File: myheroesinkotlin/app/src/main/res/drawable/app/src/main/java/com/example/myheroes/network/RetrofitInstance.java
package com.example.myheroes.network;

import android.util.Log;

import com.example.myheroes.model.HeroResults;
import com.example.myheroes.util.Constants;

import retrofit2.Call;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class RetrofitInstance {

    private final String BASE_URL = Constants.BASE_URL;
    private Service service;

    public RetrofitInstance() {
        service = createService( getInstance() );
    }


    private Retrofit getInstance() {

        return new Retrofit
                .Builder()
                .baseUrl( BASE_URL )
                .addConverterFactory( GsonConverterFactory.create() )
                .build();

    }

    private Service createService(Retrofit retrofit) {

        return retrofit.create( Service.class );

    }

    public Call<HeroResults> getRepositories() {

        Log.d( "tag", "getRepositories: " + service.getHeroes( "Spiderman", Constants.KEY ) );

        return service.getHeroes( "Spiderman", Constants.KEY );

    }

    public Call<HeroResults> getRepositories(String hero) {

        Log.d( "tag", "getRepositories: " + service.getHeroes( "Spiderman", Constants.KEY ) );

        return service.getHeroes( hero, Constants.KEY );

    }

    public Call<HeroResults> getHeroRepo(String movie, String key) {

        return service.getHeroes( movie, key );

    }

}

File: myheroesinkotlin/app/src/main/java/com/example/myheroesinkotlin/network/RetrofitInstance.kt
package com.example.myheroesinkotlin.network

import com.example.myheroesinkotlin.model.HeroResults
import com.example.myheroesinkotlin.util.Constants
import retrofit2.Call
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class RetrofitInstance() {


    val test = Constants.BASE_URL
    private var service: Service

    init {
        service = createService(getInstance())
    }

    private fun getInstance(): Retrofit {

        return Retrofit.Builder()
            .baseUrl(Constants.BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
    }

    private fun createService(retrofit: Retrofit): Service {
        return retrofit.create(Service::class.java)
    }

    fun getRepositories(): Call<HeroResults?>? {

        return service.getHeroes("Spiderman", Constants.KEY)
    }

    fun getRepositories(hero: String): Call<HeroResults?>? {
        return service.getHeroes(hero, Constants.KEY)
    }

}